roofit/histfactory/src/PiecewiseInterpolation.cxx

```diff
@@ -219,6 +219,7 @@ void PiecewiseInterpolation::fillCacheObject(RooAbsCachedReal::FuncCacheElem& ca
     RooFIter paramIter(_paramSet.fwdIterator()) ;
 
     while((param=(RooAbsReal*)paramIter.next())) {
+      ++i;
```

Comment:
if we increment here and `i` starts from zero we are always looking at the wrong `alpha`'s interp code

Reply:
Please change line 215 to `int i=-1;`

Reply:
The same changes are needed in `PiecewiseInterpolation::evaluate()` (same bug exists there). See lines 383 - 396